### PR TITLE
reorganize the code into different files and update the deps

### PR DESCRIPTION
**Pull request by TheBearodactyl**
gonna go on a debugging spree now, i got an open source license for the fabled rustrover ide :3

### CODE REVIEW

**Comment by TheBearodactyl** [comment]
also fixed some clippy warnings

**Comment by blyxyas** [comment]
Currently getting some bottom surgery sowwy :3

**Comment by TheBearodactyl** [comment]
> Currently getting some bottom surgery sowwy :3

good luck :D

**Comment by TheBearodactyl** [comment]
@blyxyas you good to review the pr now?

**Comment by blyxyas** [comment]
... I'll be another week in the hospital... It's a big surgery

**Comment by TheBearodactyl** [comment]
> ... I'll be another week in the hospital... It's a big surgery

oof. I'll try to make some more improvements while you're there :D

**Comment by TheBearodactyl** [comment]
MERRY CHRISTMAS :DDD

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "cuteness"
-version = "0.9.9"
+version = "0.9.91"
```
Comment (blyxyas) [inline Cargo.toml]: What is this progression :sob: 

```suggestion
version = "0.9.10"
```

File: src/config.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Config {
+    pub config: HashMap<String, Value>,
+    pub misc: MiscConfig,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct MiscConfig {
+    /// me when puro
```
Comment (blyxyas) [inline src/config.rs:17]: W-What's this doc comment 0_o

File: src/main.rs [hunk 3]
```diff
@@ -172,9 +93,7 @@ fn build(port: u16, outdir: &Path, sassbin: String) -> Result<()> {
 
         lazy_static!{
             static ref RE: regex::Regex = regex::Regex::new("([<>])").unwrap();
-        };
-
-        RE.replace_all(&path, "_").to_string()
+        }RE.replace_all(&path, "_").to_string()
```
Comment (blyxyas) [inline src/main.rs:96]: ?? This formatting is weird

File: src/main.rs [hunk 4]
```diff
@@ -217,7 +134,7 @@ fn build(port: u16, outdir: &Path, sassbin: String) -> Result<()> {
     f.read_to_string(&mut content)
         .context("Couldn't read configuration `cuteconfig.toml`")?;
 
-    let config = toml::from_str::<Config>(&content).context("Couldn't parse configuration")?;
+    let config: Config = toml::from_str::<Config>(&content).context("Couldn't parse configuration")?;
```
Comment (blyxyas) [inline src/main.rs]: If you declare the type at the local, there's no need for this


```suggestion
    let config: Config = toml::from_str(&content).context("Couldn't parse configuration")?;
```

**Review by blyxyas — CHANGES_REQUESTED** [review]
Some time ago I was a pretty heavy tab user. We should totally convert all tabs to spaces.

File: Cargo.toml (hunk 1 above)
Comment (TheBearodactyl) [reply Cargo.toml]: something that i am NOT good at lol

File: src/config.rs (hunk 2 above)
Comment (TheBearodactyl) [reply src/config.rs:17]: > W-What's this doc comment 0_o

easteregg :3

File: src/main.rs (hunk 3 above)
Comment (TheBearodactyl) [reply src/main.rs:96]: don't ask me, blame jetbrains

File: src/main.rs (hunk 4 above)
Comment (TheBearodactyl) [reply src/main.rs]: fair

File: src/main.rs (hunk 3 above)
Comment (blyxyas) [reply src/main.rs:96]: Then just don't autoformat, just remove this manually ฅ( ᵕ ω ᵕ )ฅ

```suggestion
        };
        
        RE.replace_all(&path, "_").to_string()
```

File: src/main.rs [hunk 5]
```diff
@@ -184,9 +103,7 @@ fn build(port: u16, outdir: &Path, sassbin: String) -> Result<()> {
     handlebars_helper!(is_pure: |src: String| {
         lazy_static!{
             static ref RE: regex::Regex = regex::Regex::new("([<>])").unwrap();
-        };
-
-        !RE.is_match(&src)
+        }!RE.is_match(&src)
```
Comment (blyxyas) [inline src/main.rs]:
```suggestion
        };
        
        !RE.is_match(&src)
```

**Comment by TheBearodactyl** [comment]
should be good to merge now

**Review by blyxyas — APPROVED** [review]
LGTM :shipit:

**Pull request merged**